### PR TITLE
Speed up GHA Python workflows

### PR DESCRIPTION
**Pull request by max-sixty**

### CODE REVIEW

**Comment by max-sixty** [comment]
The diff isn't that clear, because my autoformatter changed the YAML, and I renamed the file to `python` from `publish-python` given it also acts as a test. But this basically adds a cache to the steps in an attempt to close https://github.com/prql/prql/issues/391

**Comment by max-sixty** [comment]
Cache doesn't seem to work, so instead skipping runs where no files in `prql-python` are changed

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    if: "startsWith(github.ref, 'refs/tags/')"
+    needs: [macos, windows, linux]
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: wheels
+      - name: Publish to PyPI
+        uses: messense/maturin-action@v1
+        env:
+          MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (max-sixty) [inline .github/workflows/python.yml:90]: @charlie-sanders does this exist?

Comment (manning79) [reply .github/workflows/python.yml:90]: @max-sixty It doesn't on this repo no.  On my fork I went to Settings to add the Secret, but I don't see a settings tab in the organizational repo ?

We can also move this under the prql/ umbrella now, if you want to create a PyPi account for prql and use that

**Pull request merged**